lib/browser.js

```diff
@@ -35,14 +35,21 @@ function Browser (mdns, opts, onup) {
   this._mdns = mdns
   this._onresponse = null
   this._serviceMap = {}
+  this._txt = null
 
   if (!opts || !opts.type) {
     this._name = WILDCARD
     this._wildcard = true
+    this._txt = dnsTxt()
   } else {
     this._name = serviceName.stringify(opts.type, opts.protocol || 'tcp') + TLD
     if (opts.name) this._name = opts.name + '.' + this._name
     this._wildcard = false
+    if (!opts.txt || !opts.txt.binary) {
```

Comment:
In case the `txt` property is given, why not just pass it into the `dnsTxt` constructor as it is?